### ISSUE TITLE
training problem

### ISSUE THREAD

**Issue by RuoLan-17** [EvilGhostY/SMRSSeg#2]
The accuracy of the background class in training EIGnet is resulting in NaN.

**Comment by EvilGhostY** [comment]
Hello, if you are using the training and testing framework from [GeoSeg](https://github.com/WangLibo1995/GeoSeg), I haven't encountered any NaN issues with the clutter class during training or testing. If you are using a different training framework, it might lead to NaN values. In other training frameworks, the clutter class might be underrepresented, but training for more epochs could improve accuracy. Also, if you're using the ISPRS datasets, most papers typically don't discuss the accuracy of the clutter class.

**Comment by RuoLan-17** [comment]
![image](https://github.com/user-attachments/assets/a5ca478e-316a-48c6-a404-e8931bb8f02b)
Hello, I am also using the GeoSeg framework for training on the Vaihingen dataset. During training, in addition to the clutter class, the car class also encounters NaN values in some metrics at certain points, although the NaN values disappear by the end of training. I'm a bit confused because this issue didn't occur when training other models.

**Comment by EvilGhostY** [comment]
Hello, as shown in the figure, the issue of NaN values does indeed occur. However, if you are using the GeoSeg framework, this problem should not arise. During our testing and multiple repetitions of the experiments when we uploaded the code, we did not encounter the issue you mentioned. Of course, we are also aware that the GeoSeg training framework has been updated recently, so I have just uploaded my previous training function. You can try it again. Additionally, regarding the appearance of NaN values, I suspect that the issue may be related to the calculation of the confusion matrix. You could debug the relevant code to investigate further.

**Comment by RuoLan-17** [comment]
Thank you very much for your help. I just realized that the issue might be due to this: in my code, I have self.boundary_loss = BoundaryLoss(n_classes=6, radius=1, alpha=0.01), while in the GitHub version, it is set to self.boundary_loss = BoundaryLoss(n_classes=1, radius=1, alpha=0.01).

**Comment by EvilGhostY** [comment]
Wish you success in your research. if you have any questions, and I will respond promptly.
